### DISCUSSION TITLE
Component rendered multiple times on initial fetch?

### DISCUSSION THREAD

**Discussion by axhamre** [TanStack/query#401] [General] [answered]
On initial request ('get'), the component is re-rendered 4 times for `loading` state and then another 4 times when the `data` is at hand.

I'm using the simplest possible setup (taken from the basic example).

What could be the reason for the multiple re-rendering? And is there any way to prevent it?

**Comment by tannerlinsley** [comment, marked as answer]
- Make sure you've read over the ["Defaults to keep in mind" section in the docs](https://github.com/tannerlinsley/react-query#defaults-to-keep-in-mind)
- React Query is managing a lot of state under the hood. This includes fetching, promises, errors, etc, to manage that state appropriately, the hook must rerender.
- If you're still concerned, please create a codesandbox that shows the behavior and how you think it may be a bug
- Also this: https://twitter.com/tannerlinsley/status/1253000896776073221

**Reply by axhamre** [reply]
I see, thank you!
